### ISSUE TITLE
No stdout from background thread with prompt-toolkit>=2

### ISSUE THREAD

**Issue by topherocity** [ipython/ipython#12021]
It's always possible that I'm doing something outside of the realm of what is supposed to be expected behavior, but behavior of my code has changed after the release of `prompt-toolkit v3`. 

Before, a section of my code printed status updates from a background thread to stdout. Following the release of prompt-toolkit 3, that functionality no longer exists. Take for example a piece of minimal code (let's call it `test.py`):

```python
import time
import threading


def test_stdout():
    while(True):
        print('test hello!')
        time.sleep(1)


threading.Thread(target=test_stdout, daemon=True).start()
```

The behavior I expect with `ipython -i test.py` is:

```
Python 3.6.6 (default, Aug 16 2018, 14:51:26)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.10.1 -- An enhanced Interactive Python. Type '?' for help.
test hello!

test hello!
test hello!
test hello!
In [1]:
```

I can confirm that this output is still the case with `prompt-toolkit<3`. With the latest version of `prompt-toolkit==3.0.2`, the output after many of seconds of time is:

```
Python 3.6.6 (default, Aug 16 2018, 14:51:26)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.10.1 -- An enhanced Interactive Python. Type '?' for help.
test hello!

In [1]:
``` 

Here's my IPython environment:

```
{'commit_hash': 'e508b6e4d',
 'commit_source': 'installation',
 'default_encoding': 'UTF-8',
 'ipython_path': '/Users/choward36/.pyenv/versions/3.6.6/envs/ptk2/lib/python3.6/site-packages/IPython',
 'ipython_version': '7.10.1',
 'os_name': 'posix',
 'platform': 'Darwin-18.7.0-x86_64-i386-64bit',
 'sys_executable': '/Users/choward36/.pyenv/versions/3.6.6/envs/ptk2/bin/python',
 'sys_platform': 'darwin',
 'sys_version': '3.6.6 (default, Aug 16 2018, 14:51:26) \n'
                '[GCC 4.2.1 Compatible Clang 3.9.1 (tags/RELEASE_391/final)]'}
```

**Comment by Carreau** [comment]
Would yo mind testing with master ? That should be fixed.

**Comment by topherocity** [comment]
Yes! This appears to be fixed in 7.11.0.dev0 installed from master.
